app/controllers/subscriptions_controller.rb

```diff
@@ -15,9 +15,12 @@ def load_user
   # GET /subscriptions.xml
   def index
     @subscriptions = @user.subscriptions.includes(:subscribable)
-    if params[:type].present?
+
+    if params[:type] && Subscription::VALID_SUBSCRIBABLES.include?(params[:type].singularize.titleize)
+      @subscribable_type = params[:type]
       @subscriptions = @subscriptions.where(subscribable_type: params[:type].classify)
```

Comment:
Now that we have `@subscribable_type`, it would probably make sense to use that in place of `params[:type]` wherever possible (so here and in various places in the view file).